### PR TITLE
#14406: Add CCL Perf tests to pipeline 

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Ticket
#14406 

### Problem description
Need CCL Perf to be triggered in pipeline

### What's changed
Adds CCL Perf tests to T3K perf pipeline

T3K Model Perf - https://github.com/tenstorrent/tt-metal/actions/runs/11718145347/job/32639080900
Perf Artifact - https://github.com/tenstorrent/tt-metal/actions/runs/11718145347/job/32639080900#step:13:33

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by SeanNijjar — APPROVED** [review]
🤩

**Review by blozano-tt — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

**Review by afuller-TT — APPROVED** [review]

File: .github/workflows/t3000-model-perf-tests.yaml [hunk 1]
```diff
@@ -11,7 +11,13 @@ jobs:
...
     secrets: inherit
+  build-artifact-profiler:
+    uses: ./.github/workflows/build-artifact.yaml
+    with:
+      arch: '["wormhole_b0"]'
+      tracy: true
+    secrets: inherit
   t3000-model-perf-tests:
-    needs: build-artifact
+    needs: [build-artifact, build-artifact-profiler]
```
Comment (ttmchiou) [inline .github/workflows/t3000-model-perf-tests.yaml:21]: Can we just use only the build-artifact-profiler here?
I don't think we need build-artifact anymore if we want to use profiler tools here.

**Review by jvegaTT — APPROVED** [review]
Looks good

File: .github/workflows/t3000-model-perf-tests.yaml (hunk 1 above)
Comment (Aswinmcw) [reply .github/workflows/t3000-model-perf-tests.yaml:21]: I have introduced an if statement in the impl file to choose which build to use for that job

**Review by ttmchiou — APPROVED** [review]

**Pull request merged**